### ISSUE TITLE
sqlite3 and -PIC

### ISSUE THREAD

**Issue by All8Up** [ruslo/hunter#1580]
I'm linking sqlite3 into an so on Linux and can't seem to get the config to work correctly.  I'm probably doing this wrong but the following basic pattern worked for other libraries so I'm at a bit of a loss:

hunter_config (sqlite3 VERSION 3.24.0-p0 CMAKE_ARGS CMAKE_POSITION_INDEPENDENT_CODE=TRUE)

This generates the following:
[hunter] SQLITE3_ROOT: C:/.hunter/_Base/acbf4b9/20c3b90/1810335/Install (ver.: 3.24.0-p0)
[hunter ** INTERNAL **] Unknown error
  code: 3
  message: "URL using bad/illegal format or missing URL"
[hunter ** INTERNAL **] [Directory:C:/.hunter/_Base/Download/Hunter/0.23.33/acbf4b9/Unpacked/cmake/projects/sqlite3]

Suggestions would be welcome, I'm not much of a Linux build person and I find the whole -PIC thing a serious PITA..  :)

**Comment by ruslo** [comment]
> I'm linking sqlite3 into an so on **Linux**

> [hunter] SQLITE3_ROOT: **C:/.hunter/**_Base/acbf4b9/20c3b90/1810335/Install (ver.: 3.24.0-p0)

Looks like a Windows path is used, are you sure about Linux?

> hunter_config (sqlite3 VERSION 3.24.0-p0 CMAKE_ARGS CMAKE_POSITION_INDEPENDENT_CODE=TRUE)

See https://docs.hunter.sh/en/latest/overview/customization/toolchain-id.html#simple-toolchains

> message: "URL using bad/illegal format or missing URL"

Works fine for me, please provide minimal example to reproduce the issue. Basically you should follow this instructions while reporting bugs: https://docs.hunter.sh/en/latest/contributing.html#reporting-bugs

**Comment by All8Up** [comment]
Yup, I was testing on Windows after seeing the error in my continuous integration server.  It didn't seem to matter which build I tried, adding the config with the pic flag caused all three targets to fail.  I was hoping this was something quick and I had misunderstood the usage of hunter_config, now that that does not seem to be the problem, I'll build up a quick testbed I can post or figure out what I'm doing incorrectly in other areas.

**Comment by All8Up** [comment]
Hmph, not sure about this yet but the problem seems to be an order issue.  If I use hunter_add_package for any of my repos 'before' the addition of sqlite3, the problem occurs.  If I simply move the sqlite3 up above those internal packages it seems to work.  I.e.
fails:
hunter_add_package (mystuff)  find_package (mystuff CONFIG REQUIRED)
hunter_add_package (sqlite3) find_package (sqlite3 CONFIG REQUIRED)
works:
hunter_add_package (sqlite3) find_package (sqlite3 CONFIG REQUIRED)
hunter_add_package (mystuff)  find_package (mystuff CONFIG REQUIRED)

I will attempt to figure this out in more detail a bit later, I have to fix up some other items to be able to verify this in more detail.
